extensions/rich_text_components/components.py

```diff
@@ -122,6 +122,15 @@ def validate(cls, value_dict):
 class Link(BaseRteComponent):
     """Class for Link component."""
 
+    @classmethod
+    def validate(cls, value_dict):
+        """Validates Link component."""
+        super(Link, cls).validate(value_dict)
+        regex_pattern = r'.*mailto:.*'
+        url_with_value = value_dict['url-with-value']
+        if re.match(regex_pattern, url_with_value):
```

Comment:
Shouldn't we remove this code since the validator is already added [here](https://github.com/oppia/oppia/pull/8822/files#diff-1a892bc24b6c18dcd3edb68a2904cbc8R330)? Since the validator does the same thing, it is better to have a single source of truth.

For validation:

- `normalize` is called [here](https://github.com/oppia/oppia/blob/develop/extensions/rich_text_components/components.py#L87) to validate. 

- This will next call the `normalize_against_schema` function [here](https://github.com/oppia/oppia/blob/develop/extensions/objects/models/objects.py#L61).

- Now, the `normalize_against_schema` function will check for the validators automatically due to this [line](https://github.com/oppia/oppia/blob/develop/schema_utils.py#L161).